ros/src/computing/planning/decision/packages/decision_maker/nodes/decision_maker/decision_maker_node_state_mission.cpp

```diff
@@ -0,0 +1,196 @@
+  }
+}
+
+void DecisionMakerNode::updateWaitOrderState(cstring_t& state_name, int status)
+{
+  if (isEventFlagTrue("received_based_lane_waypoint"))
+  {
+    setEventFlag("received_based_lane_waypoint", false);
+    tryNextState("received_mission_order");
+  }
+}
+void DecisionMakerNode::exitWaitOrderState(cstring_t& state_name, int status)
+{
+}
+
+void DecisionMakerNode::entryMissionCheckState(cstring_t& state_name, int status)
+{
+  publishOperatorHelpMessage("Received mission, checking now...");
+  setEventFlag("received_back_state_waypoint", false);
+
+  int gid = 0;
+  for (auto& lane : current_status_.based_lane_array.lanes)
+  {
+    int lid = 0;
+    for (auto& wp : lane.waypoints)
+    {
+      wp.wpstate.aid = 0;
+      wp.wpstate.steering_state = autoware_msgs::WaypointState::NULLSTATE;
+      wp.wpstate.accel_state = autoware_msgs::WaypointState::NULLSTATE;
+      wp.wpstate.stop_state = autoware_msgs::WaypointState::NULLSTATE;
+      wp.wpstate.lanechange_state = autoware_msgs::WaypointState::NULLSTATE;
+      wp.wpstate.event_state = 0;
+      wp.gid = gid++;
+      wp.lid = lid++;
+      if (!isEventFlagTrue("received_back_state_waypoint") && wp.twist.twist.linear.x < 0.0)
+      {
+        setEventFlag("received_back_state_waypoint", true);
+        publishOperatorHelpMessage("Received back waypoint.");
+      }
+    }
+  }
+
+  // waypoint-state set and insert interpolation waypoint for stopline
+  setWaypointState(current_status_.based_lane_array);
+
+  // indexing
+  gid = 0;
+  for (auto& lane : current_status_.based_lane_array.lanes)
+  {
+    int lid = 0;
+    for (auto& wp : lane.waypoints)
+    {
+      wp.gid = gid++;
+      wp.lid = lid++;
+    }
+  }
+
+  current_status_.using_lane_array = current_status_.based_lane_array;
+  Pubs["lane_waypoints_array"].publish(current_status_.using_lane_array);
+  if (!isSubscriberRegistered("final_waypoints"))
+  {
+    Subs["final_waypoints"] =
+        nh_.subscribe("final_waypoints", 100, &DecisionMakerNode::callbackFromFinalWaypoint, this);
+  }
+}
+void DecisionMakerNode::updateMissionCheckState(cstring_t& state_name, int status)
+{
+  if (isEventFlagTrue("received_finalwaypoints") && current_status_.closest_waypoint != -1)
+  {
+    tryNextState("mission_is_compatible");
+  }
+}
+
+void DecisionMakerNode::entryMissionAbortedState(cstring_t& state_name, int status)
+{
+  tryNextState("operation_end");
+}
+void DecisionMakerNode::updateMissionAbortedState(cstring_t& state_name, int status)
+{
+  if (!use_management_system_)
+  {
+    sleep(1);
+    tryNextState("goto_wait_order");
+    return;
+  }
+}
+
+void DecisionMakerNode::entryDriveReadyState(cstring_t& state_name, int status)
+{
+  publishOperatorHelpMessage("Please order to engage");
+}
+
+void DecisionMakerNode::updateDriveReadyState(cstring_t& state_name, int status)
+{
+  const bool start_flag = false;
+  if (start_flag /*isEventFlagTrue("")*/)
```

Comment:
This process is not executed, but are you leaving it?

Reply:
Ok, I'll remove it.